### PR TITLE
SONAR-6258 Persist symbols into file sources

### PR DESCRIPTION
**Pull request by julienlancelot**
SONAR-6258 Persist symbols into file sources

### CODE REVIEW

**Comment by GithubIntegration** [comment]
SonarQube analysis reported 19 new issues:
* 1 blocking
* 12 major
* 1 minor
* 5 info

**Comment by GithubIntegration** [comment]
SonarQube analysis reported 4 new issues:
* 3 major
* 1 info

File: server/sonar-server/src/main/java/org/sonar/server/computation/source/SymbolsLineReader.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      }
+    }
+    return lineSymbols;
+  }
+
+  private static boolean matchLine(BatchReport.Range range, int line) {
+    return range.getStartLine() <= line && range.getEndLine() >= line;
+  }
+
+  private void sortSymbols() {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/computation/source/SymbolsLineReader.java]: where can I find the old way or sorting ?

File: sonar-batch/src/test/java/org/sonar/batch/index/SourceDataFactoryTest.java [hunk 2]
```diff
@@ -364,6 +364,46 @@ public void applySymbolReferences() throws Exception {
   }
 
   @Test
+  public void applySymbolReferences2() throws Exception {
```
Comment (teryk) [inline sonar-batch/src/test/java/org/sonar/batch/index/SourceDataFactoryTest.java]: would it be possible to have more meaningfull name for this method and the next ?

**Comment by teryk** [comment]
any idea about the new sq issues ? 
I think it would important to have some performance test before persisting file_sources on server side (just a memo).

**Comment by teryk** [comment]
apply minor feedback and LGTM !

File: server/sonar-server/src/main/java/org/sonar/server/computation/source/SymbolsLineReader.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/computation/source/SymbolsLineReader.java]: What do you mean ?

File: sonar-batch/src/test/java/org/sonar/batch/index/SourceDataFactoryTest.java (hunk 2 above)
Comment (julienlancelot) [inline sonar-batch/src/test/java/org/sonar/batch/index/SourceDataFactoryTest.java]: Good point !

**Comment by julienlancelot** [comment]
For SQ issues, as I've rebased from the master, I'll have a look.
For perf test, it will be indeed needed to do it.

File: server/sonar-server/src/main/java/org/sonar/server/computation/source/SymbolsLineReader.java (hunk 1 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/computation/source/SymbolsLineReader.java]: I wanted to compare with batch way of sorting to check if it's the same ordering

**Comment by julienlancelot** [comment]
@teryk Here is where the batch sort symbols : https://github.com/SonarSource/sonarqube/blob/master/sonar-batch/src/main/java/org/sonar/batch/index/SourceDataFactory.java#L250

**Comment by teryk** [comment]
thanks, so I confirm my LGTM

**Pull request closed without merging**